src/driver-gskill.c

```diff
@@ -1113,6 +1113,9 @@ gskill_update_resolutions(struct ratbag_profile *profile)
 			&report->dpi_levels[report->dpi_num];
 
 		res = ratbag_profile_get_resolution(profile, i);
+		if (!res->dirty)
+			continue;
+
```

Comment:
gskill starts by clearing the memory allocated with the resolutions. I am sure you are basically resetting the DPI now. gskill shouldn't check for the dirty flag.